Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -607,27 +607,69 @@ def merge_json_files(json_files)
   def merge(args:)
     bottles_hash = merge_json_files(parse_json_files(args.named))
 
-    # TODO: deduplicate --no-json bottles by:
-    # 1. throwing away bottles for newer versions of macOS if their SHA256 is
-    #    identical
-    # 2. generating `all: $SHA256` bottles that can be used on macOS and Linux
-    #    i.e. need to be `any_skip_relocation` and contain no ELF/MachO files.
-
     any_cellars = ["any", "any_skip_relocation"]
     bottles_hash.each do |formula_name, bottle_hash|
       ohai formula_name
 
       bottle = BottleSpecification.new
       bottle.root_url bottle_hash["bottle"]["root_url"]
       bottle.rebuild bottle_hash["bottle"]["rebuild"]
+
+      # if all the cellars and checksums are the same: we can create an
+      # `all: $SHA256` bottle.
+      all_bottle = bottle_hash["bottle"]["tags"].values.uniq do |tag_hash|
+        "#{tag_hash["cellar"]}-#{tag_hash["sha256"]}"
+      end.count == 1
```

Comment:
I suggest only doing this if we have more than one tag to begin with.

Linux only formulae, for example, will only have a single tag.

Reply:
@Bo98 Good catch, agreed.